src/hotspot/cpu/riscv/riscv.ad

```diff
@@ -1815,6 +1816,8 @@ const bool Matcher::match_rule_supported(int opcode) {
   }
 
   switch (opcode) {
+    case Op_OnSpinWait:
+      return VM_Version::supports_on_spin_wait();
```

Comment:
You can use `UseZihintpause` directly just like for `UseRVV`, `UsePopCountInstruction`, or `UseZbb` below.

Reply:
supports_on_spin_wait() is called both from share and cpu specific code. I think it's better to be a unified switch.